databuilder/backends/tpp.py

```diff
@@ -117,11 +117,17 @@ class TPPBackend(BaseBackend):
                     ELSE 0
                 END AS has_postcode,
                 CASE
-                    WHEN carehm.PatientAddress_ID IS NULL THEN 1
+                    WHEN carehm.PatientAddress_ID IS NOT NULL THEN 1
                     ELSE 0
                 END AS care_home_is_potential_match,
-                carehm.LocationRequiresNursing AS care_home_requires_nursing,
-                carehm.LocationDoesNotRequireNursing AS care_home_does_not_require_nursing
+                CASE
+                    WHEN carehm.LocationRequiresNursing = 'Y' THEN 1
+                    WHEN carehm.LocationRequiresNursing = 'N' THEN 0
+                 END AS care_home_requires_nursing,
+                CASE
+                    WHEN carehm.LocationDoesNotRequireNursing = 'Y' THEN 1
+                    WHEN carehm.LocationDoesNotRequireNursing = 'N' THEN 0
+                 END AS care_home_does_not_require_nursing
```

Comment:
Yay tests.

Did this creep in from a mistranslation of the cohort-extractor code?  Or is the bug present there too?

Reply:
It's not present in cohortextractor. It's not even really a mistranslation as there's no direct equivalent in cohortextractor. It was just new code I introduced to make the table "API" nicer, but I was obviously wasn't concentrating very hard at the time.